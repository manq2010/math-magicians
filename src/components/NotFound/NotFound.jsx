import React from 'react';
import './NotFound.css';

function NotFound() {
  return (
    <div className="not-found">
      Oops Page Not Found!
    </div>
  );
}

export default NotFound;
