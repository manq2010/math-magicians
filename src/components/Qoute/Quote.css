.qoutes {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 2rem auto;
}

button {
  margin: 1rem auto;
  padding: 0.5rem 1rem;
  border: none;
  border-radius: 0.5rem;
  background: #3fffd2;
  font-weight: bold;
  cursor: pointer;
}

blockquote {
  padding: 3rem;
  width: fit-content;
}
