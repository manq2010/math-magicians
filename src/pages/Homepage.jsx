import React from 'react';
import '../App.css';

function Homepage() {
  return (
    <div className="home-page">
      <h2>Welcome to our page!</h2>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Saepe, quidem?
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Consectetur perferendis
        neque illum, officiaquisquam quasi voluptas debitis possimus assumenda! Ipsum reprehenderit
      </p>

      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Saepe, quidem?
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Consectetur perferendis
        neque illum, officiaquisquam quasi voluptas debitis possimus assumenda! Ipsum reprehenderit
        maiores optio provident pariatur. Ipsum iure natus quisquam soluta!
      </p>
    </div>
  );
}

export default Homepage;
