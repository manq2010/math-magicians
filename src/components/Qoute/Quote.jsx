import React, { useState } from 'react';
import './Quote.css';

function Quote() {
  const [quote, setQuote] = useState({
    quoteText: 'Once we accept our limits, we go beyond them.',
    author: 'albert-einstein',
  });

  const randomQuate = () => {
    fetch('https://api.quotable.io/random')
      .then((response) => response.json())
      .then((data) => setQuote({ quoteText: data.content, author: data.author }));
  };

  return (
    <div className="qoutes">
      <blockquote>
        <p>
          {quote.quoteText}
          {' '}
          -
          {' '}
          {quote.author}
        </p>
      </blockquote>
      <button onClick={() => randomQuate()} type="button">Next Quote</button>
    </div>

  );
}

export default Quote;
