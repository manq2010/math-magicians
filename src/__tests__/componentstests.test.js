import { BrowserRouter, Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import NotFound from '../components/NotFound/NotFound';
import Homepage from '../pages/Homepage';
import Quote from '../components/Qoute/Quote';
import AppRoutes from '../routes';

// Routing test

describe('Navbar renders', () => {
  test('should redirect and update history', () => {
    const history = createMemoryHistory();
    render(
      <Router location={history.location} navigator={history}>
        <AppRoutes />
      </Router>,
    );

    fireEvent.click(screen.getByTestId(/Calculator/));
    expect(history.location.pathname).toEqual('/calculator');
  });

  test('should redirect and update dom', () => {
    render(
      <BrowserRouter>
        <AppRoutes />
      </BrowserRouter>,
    );

    fireEvent.click(screen.getByTestId(/Calculator/));
    expect(screen.getByText(/Let's do some math!/)).toBeInTheDocument();
  });
});

describe('Not-found page is rendered', () => {
  test('Component is rendered', () => {
    render(<NotFound />);
    const container = screen.getByText(/Oops Page Not Found!/);
    expect(container).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<NotFound />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('HomePage component is rendererd', () => {
  test('Component is rendererd', () => {
    render(<Homepage />);
    const container = screen.getByText(/Welcome to our page!/);
    expect(container).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Homepage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Qoute component is rendererd', () => {
  test('Component is rendererd', () => {
    render(<Quote />);
    const container = screen.getByTestId(/qoutes/);
    expect(container).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
