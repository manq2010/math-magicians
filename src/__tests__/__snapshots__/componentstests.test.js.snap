// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component is rendererd matches snapshot 1`] = `
<div
  className="home-page"
>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Saepe, quidem? Lorem ipsum dolor, sit amet consectetur adipisicing elit. Consectetur perferendis neque illum, officiaquisquam quasi voluptas debitis possimus assumenda! Ipsum reprehenderit
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Saepe, quidem? Lorem ipsum dolor, sit amet consectetur adipisicing elit. Consectetur perferendis neque illum, officiaquisquam quasi voluptas debitis possimus assumenda! Ipsum reprehenderit maiores optio provident pariatur. Ipsum iure natus quisquam soluta!
  </p>
</div>
`;

exports[`Not-found page is rendered matches snapshot 1`] = `
<div
  className="not-found"
>
  Oops Page Not Found!
</div>
`;

exports[`Qoute component is rendererd matches snapshot 1`] = `
<div
  className="qoutes"
>
  <blockquote>
    <p
      data-testid="qoutes"
    >
      Once we accept our limits, we go beyond them.
       
      -
       
      albert-einstein
    </p>
  </blockquote>
  <button
    onClick={[Function]}
    type="button"
  >
    Next Quote
  </button>
</div>
`;
